### PR TITLE
Add "requiredSaleorVersion" field to the app manifest

### PR DESCRIPTION
**Pull request by przlada**
I want to merge this change because it adds a new optional app manifest field: `requiredSaleorVersion`. This field should contain the `semver` range, which specifies the Saleor version required by an app to work properly or was tested with.
Example `requiredSaleorVersion` valid values:
- `>=3.6 <4`
- `^3.8`
- `3.4 - 3.8 || ^3.10`

A new optional `Manifest.requiredSaleorVersion` field of type `ManifestRequiredSaleorVersion` is added:
```graphql
type ManifestRequiredSaleorVersion {
  constraint: String!
  satisfied: Boolean!
}
```
`appFetchManifest` mutation returns it when the manifest contains `requiredSaleorVersion`. For example, it can be used by the dashboard to inform users if chosen app supports the current saleor version.
Example response:
```json
{
   "data":{
      "appFetchManifest":{
         "manifest":{
            "identifier":"saleor-complex-sample",
            "requiredSaleorVersion":{
               "constraint":"<3.11",
               "satisfied":false
            }
         },
         "errors":[]
      }
   },
   "extensions":{
      "cost":{
         "requestedQueryCost":0,
         "maximumAvailable":50000
      }
   }
}
```

- `appFetchManifest` and `appInstall` mutations return an error (code: `INVALID`) if the `requiredSaleorVersion` field has an improper value.
- `appInstall` breaks the app installation process and returns an error (code: `UNSUPPORTED_SALEOR_VERSION`) if the  Saleor version is not in range of `requiredSaleorVersion`

A new dependency is introduced: [semantic-version](https://github.com/rbarrois/python-semanticversion)

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by korycins — APPROVED** [review]
Looks good 💪 . Make sure that you will also update the docs.

File: saleor/graphql/app/types.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
     ADDED_IN_31,
     ADDED_IN_35,
     ADDED_IN_38,
+    ADDED_IN_312,
```
Comment (korycins) [inline saleor/graphql/app/types.py:19]: Should be 313. 312 is already released

**Review by korycins — APPROVED** [review]

File: saleor/graphql/app/types.py (hunk 1 above)
Comment (przlada) [reply saleor/graphql/app/types.py]: Good catch 👍 fixed

File: saleor/graphql/app/types.py [hunk 2]
```diff
@@ -192,6 +193,11 @@ def resolve_target_url(root, _info: ResolveInfo):
         return root["targetUrl"]
 
 
+class ManifestRequiredSaleorVersion(graphene.ObjectType):
+    constraint = graphene.String(required=True)
+    satisfied = graphene.Boolean(required=True)
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:198]: We should add descriptions here.

**Comment by maarcingebala** [comment]
@patrys This PR introduces a new package dependency; could you review it?

File: saleor/graphql/app/types.py (hunk 2 above)
Comment (przlada) [reply saleor/graphql/app/types.py]: Thanks! Added

**Review by patrys — APPROVED** [review]

**Comment by patrys** [comment]
Please link to a docs PR before merging.

**Comment by przlada** [comment]
Docs PR: https://github.com/saleor/saleor-docs/pull/678

**Review by szczecha — APPROVED** [review]

**Pull request merged**